dependency_management/_utils.py

```diff
@@ -103,29 +101,30 @@ def build_koji_repo_dict(crates, release):
         try:
             if release[0] != "f":
                 raise RuntimeError(
-                    'release argument must be "rawhide" or f<n> where n is an integer, was "%s"'
-                    % release
+                    f'release argument must be "rawhide" or f<n> '
+                    f'where n is an integer, was "{release}"'
                 )
 
             release_number = int(release[1:])
 
             if release_number < 34:
                 raise RuntimeError(
-                    "release number must be at least 34, was %d" % release_number
+                    f"release number must be at least 34, was {release_number}"
                 )
         except (IndexError, ValueError) as err:
             raise RuntimeError(
-                'release must be "rawhide" or f<n> where n is an integer, was "%s"'
-                % release
+                f'release must be "rawhide" or f<n> where n is an integer, was "{release}"'
             ) from err
 
-    url = "https://kojipkgs.fedoraproject.org/repos/%s/latest/x86_64/pkglist" % (
-        release if release == "rawhide" else "%s-build" % release
+    url = (
+        f"https://kojipkgs.fedoraproject.org/repos/{release}/latest/x86_64/pkglist"
```

Comment:
this one isn't correct, if release != "rawhide" you get just, eg., "f34-build", but what you need is an entire URL.

Reply:
Oh, I think I see what's going on: I need to encapsulate the `if` statement in the braces of the formatted string.

I have to figure out how to test this, but does this look like it will work?
```
    url = (
        "https://kojipkgs.fedoraproject.org/repos/"
        f"{release if release == 'rawhide' else f'{release}-build'}/latest/x86_64/pkglist"
    )
```

Reply:
Looks good to me.

Reply:
Does this look like a good test result, after pushing a commit to try the above?
```
$ COMPARE_FEDORA_VERSIONS=./compare_fedora_versions MANIFEST_PATH=../../Cargo.toml FEDORA_RELEASE=f35 IGNORE_ARGS="--ignore-category low" make -f ../../Makefile check-fedora-versions
echo "Testing that COMPARE_FEDORA_VERSIONS environment variable is set to a valid path"
Testing that COMPARE_FEDORA_VERSIONS environment variable is set to a valid path
test -e "./compare_fedora_versions"
./compare_fedora_versions --manifest-path=../../Cargo.toml --release=f35 --ignore-category low
Version 3.1.2 of crate clap is available in Fedora. Requires update in Cargo.toml
{"missing": [], "high": [], "low": []}
```

Reply:
That seems reasonable!